test/fixtures/s3-fixture/src/main/java/fixture/s3/S3HttpHandler.java

```diff
@@ -223,6 +224,9 @@ public void handle(final HttpExchange exchange) throws IOException {
                 if (delimiter != null) {
                     list.append("<Delimiter>").append(delimiter).append("</Delimiter>");
                 }
+                // Would be good to test pagination here (the only real difference between ListObjects and ListObjectsV2) but for now
+                // we return all the results at once.
```

Comment:
Should this explanation be partly phrased as a TODO?

Reply:
I hold no hope that we'll actually do it so I'm going to just leave it as a comment.